### PR TITLE
dfp: On proxying, continue the filter if the DNS cache has a host

### PR DESCRIPTION
**Pull request by abeyad**

### CODE REVIEW

File: source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc [hunk 1]
```diff
@@ -281,6 +282,10 @@ Http::FilterHeadersStatus ProxyFilter::decodeHeaders(Http::RequestHeaderMap& hea
 
     auto const& host = result.host_info_;
     latchTime(decoder_callbacks_, DNS_END);
+    if (is_proxying) {
+      ENVOY_BUG(host.has_value(), "Proxying request but not host entry in DNS cache.");
```
Comment (alyssawilk) [inline source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc:286]: not -> no?

**Review by alyssawilk — COMMENTED** [review]
LGTM modulo typo.

File: source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc (hunk 1 above)
Comment (abeyad) [reply source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc]: done

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**